checks/taskcluster/task_e2e.py

```diff
@@ -0,0 +1,240 @@
+"""
+A check to create a task and verify that its artifacts are accessible.
+"""
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict
+
+import aiohttp
+import taskcluster
+import taskcluster.aio
+
+from poucave import config, utils
+from poucave.typings import CheckResult
+
+
+logger = logging.getLogger(__name__)
+
+# List which check parameters are visible in the UI.
+EXPOSED_PARAMETERS = ["root_url", "project", "max_duration"]
+
+
+async def run(
+    root_url: str,
+    client_id: str,
+    access_token: str,
+    project: str = "taskcluster",
+    worker_type: str = "gw-ci-ubuntu-18-04",
+    output_message: str = "Hello World!",
+    max_duration: int = 120,  # fails if takes longer.
+    task_lifetime: int = 600,  # recreate tasks every 10 min.
+) -> CheckResult:
+    """
+    Check entrypoint, executed at an undeterminated frequency.
+
+    For example, with the following `config-taskcluster.toml` config file:
+
+    .. code-block:: toml
+
+        [checks.queue.task-e2e]
+        description = ""
+        module = "checks.taskcluster.task_e2e"
+        ttl = 60
+        params.root_url = "${TASKCLUSTER_ROOT_URL}"
+        params.client_id = "${TASKCLUSTER_CLIENT_ID}"
+        params.access_token = "${TASKCLUSTER_ACCESS_TOKEN}"
+
+    It can be executed from the command-line with:
+    ::
+
+        $ export TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com
+        $ export TASKCLUSTER_CLIENT_ID=project/taskcluster/temp-pete-and-matt
+        $ export TASKCLUSTER_ACCESS_TOKEN=bR67xxxxxxxxxxxxxxxxxxxxxxxxx
+        $ export CONFIG_FILE=config-taskcluster.toml
+        $ make check project=queue check=task-e2e
+
+    When running the web server (``make serve``), this check gets executed everytime
+    the associated HTTP endpoint ``http://.../checks/queue/task-e2e`` is reached.
```

Comment:
> Does that mean it only happens when someone views an element of a web page,

It means a `GET /checks/queue/task-e2e` is served. It can be from the UI, or a service like Pingdom, or something else. But Poucave is not in charge of regularly executing the checks